File: src/test/java/libldt3/model/saetze/FuzzerTest.java
package libldt3.model.saetze;

import de.egore911.fuzz.Fuzzer;
import libldt3.LdtConstants;
import libldt3.LdtReader;
import libldt3.LdtWriter;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;
import java.nio.file.Path;
import java.util.Collections;
import java.util.List;

public class FuzzerTest {

	private final Fuzzer fuzzer = new Fuzzer(new LdtCustomHandler());

	@Test
	public void auftragFuzzerTest() throws IOException {
		Auftrag auftrag = fuzzer.fuzz(Auftrag.class, "libldt3.model");
		testSatz(auftrag);
	}

	@Test
	public void befundFuzzerTest() throws IOException {
		Befund befund = fuzzer.fuzz(Befund.class, "libldt3.model");
		testSatz(befund);
	}

	@Test
	public void laborDatenpaketAbschlussFuzzerTest() throws IOException {
		LaborDatenpaketAbschluss laborDatenpaketAbschluss = fuzzer.fuzz(LaborDatenpaketAbschluss.class, "libldt3.model");
		testSatz(laborDatenpaketAbschluss);
	}

	@Test
	public void laborDatenpaketHeaderFuzzerTest() throws IOException {
		LaborDatenpaketHeader laborDatenpaketHeader = fuzzer.fuzz(LaborDatenpaketHeader.class, "libldt3.model");
		testSatz(laborDatenpaketHeader);
	}

	@Test
	public void praxisDatenpaketAbschlussFuzzerTest() throws IOException {
		PraxisDatenpaketAbschluss praxisDatenpaketAbschluss = fuzzer.fuzz(PraxisDatenpaketAbschluss.class, "libldt3.model");
		testSatz(praxisDatenpaketAbschluss);
	}

	@Test
	public void PraxisDatenpaketHeaderFuzzerTest() throws IOException {
		PraxisDatenpaketHeader praxisDatenpaketHeader = fuzzer.fuzz(PraxisDatenpaketHeader.class, "libldt3.model");
		testSatz(praxisDatenpaketHeader);
	}

	private void testSatz(Satz satz) throws IOException {
		Path file = File.createTempFile("auftrag_fuzzer_test", ".ldt").toPath();
		new LdtWriter(LdtConstants.Mode.RELAXED).write(Collections.singletonList(satz), file);
		List<Satz> saetze = new LdtReader(LdtConstants.Mode.RELAXED).read(file);
		Assertions.assertEquals(saetze.size(), 1);
	}
}

File: cs/libldt3.tests/libldt3/model/saetze/FuzzerTest.cs
using System;
using System.Collections.Generic;
using System.IO;
using de.egore911.fuzz;
using NUnit.Framework;

namespace libldt3
{
    namespace model
    {
        namespace saetze
        {
            public class FuzzerTest
            {

                readonly Fuzzer fuzzer = new Fuzzer(new LdtCustomHandler());

                [Test]
                public void auftragFuzzerTest()
                {
                    Auftrag auftrag = fuzzer.fuzz<Auftrag>(typeof(Auftrag), "libldt3.model");
                    testSatz(auftrag);
                }

                [Test]
                public void befundFuzzerTest()
                {
                    Befund befund = fuzzer.fuzz<Befund>(typeof(Befund), "libldt3.model");
                    testSatz(befund);
                }

                [Test]
                public void laborDatenpaketAbschlussFuzzerTest()
                {
                    LaborDatenpaketAbschluss laborDatenpaketAbschluss = fuzzer.fuzz<LaborDatenpaketAbschluss>(typeof(LaborDatenpaketAbschluss), "libldt3.model");
                    testSatz(laborDatenpaketAbschluss);
                }

                [Test]
                public void laborDatenpaketHeaderFuzzerTest()
                {
                    LaborDatenpaketHeader laborDatenpaketHeader = fuzzer.fuzz<LaborDatenpaketHeader>(typeof(LaborDatenpaketHeader), "libldt3.model");
                    testSatz(laborDatenpaketHeader);
                }

                [Test]
                public void praxisDatenpaketAbschlussFuzzerTest()
                {
                    PraxisDatenpaketAbschluss praxisDatenpaketAbschluss = fuzzer.fuzz<PraxisDatenpaketAbschluss>(typeof(PraxisDatenpaketAbschluss), "libldt3.model");
                    testSatz(praxisDatenpaketAbschluss);
                }

                [Test]
                public void PraxisDatenpaketHeaderFuzzerTest()
                {
                    PraxisDatenpaketHeader praxisDatenpaketHeader = fuzzer.fuzz<PraxisDatenpaketHeader>(typeof(PraxisDatenpaketHeader), "libldt3.model");
                    testSatz(praxisDatenpaketHeader);
                }

                void testSatz(Satz satz)
                {
                    string file = Path.GetTempPath() + "fuzzer_test" + Guid.NewGuid().ToString() + ".ldt";
                    new LdtWriter(LdtConstants.Mode.RELAXED).Write(new List<Satz> { satz }, file);
                    IList<Satz> saetze = new LdtReader(LdtConstants.Mode.RELAXED).Read(file);
                    Assert.That(saetze.Count, Is.EqualTo(1), file);
                }
            }
        }
    }
}